delfin/task_manager/scheduler/schedulers/telemetry/failed_performance_collection_handler.py

```diff
@@ -0,0 +1,110 @@
+CONF = cfg.CONF
+
+
+class FailedPerformanceCollectionHandler(object):
+    def __init__(self, ctx, failed_task_id, storage_id, args, job_id,
+                 retry_count, start_time, end_time):
+        self.ctx = ctx
+        self.failed_task_id = failed_task_id
+        self.retry_count = retry_count
+        self.storage_id = storage_id
+        self.job_id = job_id
+        self.args = args
+        self.start_time = start_time
+        self.end_time = end_time
+        self.task_rpcapi = task_rpcapi.TaskAPI()
+        self.scheduler_instance = scheduler.Scheduler.get_instance()
+        self.result = TelemetryJobStatus.FAILED_JOB_STATUS_INIT
+
+    @staticmethod
+    def get_instance(ctx, failed_task_id):
+        # fetch failed task info
+        failed_task = db.failed_task_get(ctx, failed_task_id)
+        # fetched task info
+        task = db.task_get(ctx, failed_task[FailedTask.task_id.name])
+        return FailedPerformanceCollectionHandler(
+            ctx,
+            failed_task[FailedTask.id.name],
+            task[Task.storage_id.name],
+            task[Task.args.name],
+            failed_task[FailedTask.job_id.name],
+            failed_task[FailedTask.retry_count.name],
+            failed_task[FailedTask.start_time.name],
+            failed_task[FailedTask.end_time.name],
+        )
+
+    def __call__(self):
+        # pull performance collection info
+        self.retry_count = self.retry_count + 1
+        try:
+            status = self.task_rpcapi.collect_telemetry(
+                self.ctx, self.storage_id,
+                PerformanceCollectionTask.__module__ + '.' +
+                PerformanceCollectionTask.__name__,
+                self.args, self.start_time, self.end_time)
+
+            if not status:
+                raise exception.TelemetryTaskExecError()
+        except Exception as e:
+            LOG.error(e)
+            msg = _("Failed to collect performance metrics for storage "
+                    "id:{0}, reason:{1}".format(self.storage_id,
+                                                six.text_type(e)))
+            LOG.error(msg)
+        else:
+            LOG.info("Successfully completed Performance metrics collection "
+                     "for storage id :{0} ".format(self.storage_id))
+            self.result = TelemetryJobStatus.FAILED_JOB_STATUS_SUCCESS
+            self._stop_task()
+            return
+
+        if self.retry_count >= TelemetryCollection.MAX_FAILED_JOB_RETRY_COUNT:
+            msg = _(
+                "Failed to collect performance metrics of task instance "
+                "id:{0} for start time:{1} and end time:{2} with "
+                "maximum retry. Giving up on "
+                "retry".format(self.failed_task_id, self.start_time,
+                               self.end_time))
+            LOG.error(msg)
+            self._stop_task()
+            return
+
+        self.result = TelemetryJobStatus.FAILED_JOB_STATUS_STARTED
```

Comment:
FAILED_JOB_STATUS_STARTED ..This seems confusing..for multiple retries, this is in-progress

Reply:
Renamed with ```FAILED_JOB_STATUS_RETRYING```